src/harness/util.py

```diff
@@ -22,6 +22,11 @@
 import uuid
 
 
+class TimeoutException(Exception):
```

Comment:
Is it actually important to have a custom exception type? If you're going to try-except around the helper function then it makes sense, and otherwise you may as well just raise a generic error with a custom error message.